ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -313,6 +318,30 @@ def advance(self, env: EnvManager) -> int:
 
         return num_steps
 
+    def join_threads(self, timeout_seconds: float = 1.0) -> None:
+        """
+        Wait for threads to finish, and merge their timer information into the main thread.
+        :param timeout_seconds:
+        :return:
+        """
+        self.kill_trainers = True
+        for t in self.trainer_threads:
+            try:
+                t.join(timeout_seconds)
+            except Exception:
+                pass
+
+        with hierarchical_timer("trainer_threads") as main_timer_node:
```

Comment:
would prefer to have this happen behind-the-scenes but can't think of a good way to do it. Doing it in the main thread seems fine.